### ISSUE TITLE
[ci] fatal: reference is not a tree

### ISSUE THREAD

**Issue by dev-zero** [cp2k/cp2k#393]
example: https://storage.googleapis.com/cp2k-ci/run-cp2k-sdbg-cf9d8f09_report.txt

This seems to happen when the `master` gets an update while the CI already started the job.

At least in [2017](https://discourse.drone.io/t/github-claims-that-merge-refs-are-undocumented-feature/1100) using the `/merge` ref was an undocumented feature not meant to be used by 3rd parties.

**Comment by oschuett** [comment]
Yes, this sometimes happens. It should however be fairly rare and you can always hit re-run.

Currently, I have three ideas for fixing this:
 1. Perform the merge myself in [run_target.sh](https://github.com/cp2k/cp2k-ci/blob/master/toolbox/run_target.sh) and [ci_entrypoint.sh](https://github.com/cp2k/cp2k/blob/master/tools/docker/scripts/ci_entrypoint.sh).
 2. Backup the merge branch somewhere (different repo or different branch name)
 3. Add a detection and auto re-run mechanism in the backend.

I don't really like any of them. 

> the /merge ref was an undocumented feature not meant to be used by 3rd parties.

The merge branch is not well documented but it gets mention e.g. [here](https://developer.github.com/v3/git/#checking-mergeability-of-pull-requests). And I'm pretty sure most other CI system use it as well.
